src/wrfda-ncdiag/wrfda_ncdiag.py

```diff
@@ -0,0 +1,407 @@
+    'ice_area_fraction': '1',
+    'surface_snow_area_fraction': '1',
+    'vegetation_area_fraction': '1',
+    'surface_temperature_where_sea': 'K',
+    'surface_temperature_where_land': 'K',
+    'surface_temperature_where_ice': 'K',
+    'surface_temperature_where_snow': 'K',
+    'surface_wind_speed': 'm s-1',
+    'surface_wind_from_direction': 'degree',
+    'leaf_area_index': '1',
+    'volume_fraction_of_condensed_water_in_soil': '1',
+    'soil_temperature': 'K',
+    'land_type_index': '1',
+    'vegetation_type_index': '1',
+    'soil_type': '1',
+    'surface_snow_thickness': 'm',
+    'humidity_mixing_ratio': '1',
+    'wind_reduction_factor_at_10m': '1',
+    'sulf': '1',
+    'bc1': '1',
+    'bc2': '1',
+    'oc1': '1',
+    'oc2': '1',
+    'dust1': '1',
+    'dust2': '1',
+    'dust3': '1',
+    'dust4': '1',
+    'dust5': '1',
+    'seas1': '1',
+    'seas2': '1',
+    'seas3': '1',
+    'seas4': '1',
+    'latitude': 'degrees_north',
+    'longitude': 'degrees_east',
+    'station_elevation': 'm',
+    'height': 'm',
+    'height_above_mean_sea_level': 'm',
+    'cloud_area_fraction': '1',
+    'scan_position': '1',
+    'sensor_azimuth_angle': 'degree',
+    'sensor_zenith_angle': 'degree',
+    'sensor_view_angle': 'degree',
+    'solar_zenith_angle': 'degree',
+    'solar_azimuth_angle': 'degree',
+    'modis_deep_blue_flag': '1',
+    'row_anomaly_index': '1',
+    'top_level_pressure': 'Pa',
+    'bottom_level_pressure': 'Pa',
+    'tropopause_pressure': 'Pa',
+    'brightness_temperature_jacobian_surface_temperature': '1',
+    'brightness_temperature_jacobian_surface_emissivity': 'K',
+    'brightness_temperature_jacobian_air_temperature': '1',
+    'brightness_temperature_jacobian_humidity_mixing_ratio': 'K/g/Kg ',
+    'optical_thickness_of_atmosphere_layer': '1',
+}
+
+# @TestReference
+# fields from WRFDA to compare to computations done in UFO
+test_fields = {
+}
+
+###############################################################################
+###############################################################################
+
+
+# satellite radiance observations
+class Radiances:
+    """ class Radiances - satellite radiance observations
+
+                Use this class to read in satellite radiance observations
+                from WRFDA netCDF diag files
+
+    Functions:
+
+    Attributes:
+      filename    - string path to file
+      validtime   - datetime object of valid observation time
+      nobs        - number of observations
+
+
+  """
+
+    def __init__(self, filename):
+        self.filename = filename
+        splitfname = self.filename.split('/')[-1].split('_')
+        i = False
+        for s in rad_platform_sensor_combos:
+            if s in splitfname:
+                self.platform_sensor = s
+                i = splitfname.index(s)
+        if not i:
+            raise ValueError("Observation is not a radiance type...")
+
+    def read(self):
+        import netCDF4 as nc
```

Comment:
Why do imports here as opposed to global scope?  It seems NetCDF4 will always be available.

Reply:
I moved the imports to global.  If that is the preferred convention, then other converters could be updated too.